lib/rules/prefer-async-await.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview Prefer async/await over .then()
+ * @author Elad Shahar
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determine if a node is calling a `then` property
+ * @private
+ * @param {ASTNode} node - node to test
+ * @returns {boolean} True if `node` is a `then` property call
+ */
+function isThenCall(node) {
```

Comment:
Can you rename this function to something like `isThenPropertyAccess`? If I'm understanding correctly, this will report all accesses to a `then` property, even if no method is called. For example, this will report an error:

```js
const foo = bar.then;
```

(I think this behavior from the rule is probably reasonable, but it doesn't match the current function name `isThenCall`.)